data/src/main/java/com/microsoft/azure/kusto/data/auth/UserPromptTokenProvider.java

```diff
@@ -0,0 +1,90 @@
+
+    static {
+        URI tmp;
+        try {
+            tmp = new URI("http://localhost");
+        } catch (URISyntaxException e) { // This cannot happen, but is necessary to allow for a static final variable whose instantiation can throw an exception
+            tmp = null;
+        }
+        redirectUri = tmp;
+    }
+
+    private final String usernameHint;
+
+    UserPromptTokenProvider(@NotNull String clusterUrl, String authorityId) throws URISyntaxException {
+        this(null, clusterUrl, authorityId);
+    }
+
+    UserPromptTokenProvider(String usernameHint, @NotNull String clusterUrl, String authorityId) throws URISyntaxException {
+        super(clusterUrl, authorityId);
+        this.usernameHint = usernameHint;
+
+        try {
+            clientApplication = PublicClientApplication.builder(CLIENT_ID).authority(aadAuthorityUrl).build();
+        } catch (MalformedURLException e) {
+            throw new URISyntaxException(aadAuthorityUrl, ERROR_INVALID_AUTHORITY_URL);
+        }
+    }
+
+    @Override
+    protected IAuthenticationResult acquireNewAccessToken() throws DataServiceException, DataClientException {
+        IAuthenticationResult result;
+        try {
+            // This is the only auth method that allows the same application to be used for multiple distinct accounts, so reset account cache between sign-ins
+            clientApplication = PublicClientApplication.builder(CLIENT_ID).authority(aadAuthorityUrl).build();
+            CompletableFuture<IAuthenticationResult> future = clientApplication.acquireToken(InteractiveRequestParameters.builder(redirectUri).scopes(scopes).loginHint(usernameHint).build());
+            result = future.get(USER_PROMPT_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        } catch (MalformedURLException e) {
+            throw new DataClientException(clusterUrl, ERROR_INVALID_AUTHORITY_URL, e);
+        } catch (TimeoutException | ExecutionException e) {
+            throw new DataServiceException(clusterUrl, ERROR_ACQUIRING_APPLICATION_ACCESS_TOKEN, e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new DataServiceException(clusterUrl, ERROR_ACQUIRING_APPLICATION_ACCESS_TOKEN, e);
+        }
+        if (result == null) {
+            throw new DataServiceException(clusterUrl, "acquireWithUserPrompt got 'null' authentication result");
+        }
+        return result;
+    }
+
+    @Override
+    IAccount getAccount(Set<IAccount> accountSet) {
+        if (StringUtils.isNotBlank(usernameHint)) {
+            return accountSet.stream().filter(u -> usernameHint.equalsIgnoreCase(u.username())).findAny().orElse(null);
+        } else {
+            if (accountSet.isEmpty()) {
+                return null;
+            } else {
+                // Normally we would filter accounts by the user authenticating, but there's only 1 per AadAuthenticationHelper instance
+                return accountSet.iterator().next();

```

Comment:
One or None.
First is not a good choice as it is random.

Reply:
There can only be 0 or 1 accounts in the UserPromptTokenProvider. More isn't possible by design.

Reply:
A) you can be prompted again and provide a different user 
B) security is all about everyone making sure their code is as safe as possible, so there's no harm in checking there's only one account

Reply:
Per my comment in UserPromptTokenProvider, I reset the PublicCLientApp to be sure in terms of security.

Reply:
API description strings are not comments :)

Reply:
Sorry I don't follow. I believe it's an actual comment.

Reply:
Many IDE's treat specific comments as an ad-hoc documentation if they have the right format and / or are in the right location, e.g. the /// Summary of C++

Reply:
Sorry, I'm still not following.

I wrote: "Per my comment..."
You responded: "API description strings are not comments", which implies to me that you think I made a Javadoc API description.
So I responded: "I believe it's an actual comment."

My comment is as follows:
// This is the only auth method that allows the same application to be used for multiple distinct accounts, so reset account cache between sign-ins

Are you saying you *want* it to be a Javadoc?

Reply:
Mixed up two conversations.
You can close this one